src/NLog/Internal/FilePathLayout.cs

```diff
@@ -164,6 +164,13 @@ private string GetRenderedFileName(LogEventInfo logEvent, System.Text.StringBuil
 
             if (reusableBuilder != null)
             {
+                if (!_layout.IsThreadAgnostic)
```

Comment:
I'm doubting if this should be `if (_layout.IsThreadAgnostic)`, as ThreadAgnostic stuff could be calculated on every thread. (and thus could be lazy). 

non-ThreadAgnostic should be calculated on the main thread.